packages/dht/src/connection/websocket/AbstractWebsocketClientConnection.ts

```diff
@@ -1,89 +1,47 @@
 // we have a custom counterpart
 export const CUSTOM_GOING_AWAY = 3001
+// https://github.com/websockets/ws/blob/master/doc/ws.md#ready-state-constants
+const OPEN = 1
 
-const BINARY_TYPE = 'arraybuffer'
+const logger = new Logger(module)
 
-export class ClientWebsocket extends EventEmitter<ConnectionEvents> implements IConnection {
+export abstract class AbstractWebsocketClientConnection extends EventEmitter<ConnectionEvents> implements IConnection {
 
     public readonly connectionId: ConnectionID
-    private socket?: Websocket
+    protected abstract socket?: Socket
     public connectionType = ConnectionType.WEBSOCKET_CLIENT
-
-    private destroyed = false
-
+    protected destroyed = false
+    
     constructor() {
         super()
         this.connectionId = createRandomConnectionId()
     }
 
-    // TODO explicit default value for "selfSigned" or make it required
-    public connect(address: string, selfSigned?: boolean): void {
-        if (!this.destroyed) {
-            this.socket = new Websocket(address, undefined, undefined, undefined, { rejectUnauthorized: !selfSigned })
-            this.socket.binaryType = BINARY_TYPE
-            this.socket.onerror = (error: Error) => {
-                if (!this.destroyed) {
-                    logger.trace('WebSocket Client error: ' + error?.message, { error })
-                    this.emit('error', error.name)
-                }
-            }
-
-            this.socket.onopen = () => {
-                if (!this.destroyed) {
-                    logger.trace('WebSocket Client Connected')
-                    if (this.socket && this.socket.readyState === this.socket.OPEN) {
-                        this.emit('connected')
-                    }
-                }
-            }
-
-            this.socket.onclose = (event: ICloseEvent) => {
-                if (!this.destroyed) {
-                    logger.trace('Websocket Closed')
-                    this.doDisconnect(event.code, event.reason)
-                }
-            }
-
-            this.socket.onmessage = (message: IMessageEvent) => {
-                if (!this.destroyed) {
-                    if (typeof message.data === 'string') {
-                        logger.debug('Received string: \'' + message.data + '\'')
-                    } else {
-                        this.emit('data', new Uint8Array(message.data))
-                    }
-                }
-            }
-        } else {
-            logger.debug('Tried to connect() a stopped connection')
-        }
-    }
-
-    private doDisconnect(code?: number, reason?: string) {
-        this.destroyed = true
-        this.stopListening()
-        this.socket = undefined
-        const gracefulLeave = (code === GOING_AWAY) || (code === CUSTOM_GOING_AWAY)
-        this.emit('disconnected', gracefulLeave, code, reason)
-        this.removeAllListeners()
-    }
+    public abstract connect(address: string, selfSigned?: boolean): void
```

Comment:
Keep the comment:
`"// TODO explicit default value for "selfSigned" or make it required"`

Reply:
https://github.com/streamr-dev/network/pull/2092/files